Update application-proxy-configure-api.md (#9463)
microsoftgraph/microsoft-graph-docs: concepts/application-proxy-configure-api.md

|[servicePrincipals](https://docs.microsoft.com/graph/api/resources/serviceprincipal?view=graph-rest-1.0)|[Create servicePrincipal](https://docs.microsoft.com/graph/api/serviceprincipal-post-serviceprincipals?view=graph-rest-beta&tabs=http) <br> [Update servicePrincipal](https://docs.microsoft.com/graph/api/serviceprincipal-update?view=graph-rest-1.0&tabs=http) <br> [Create appRoleAssignments](https://docs.microsoft.com/graph/api/serviceprincipal-post-approleassignments?view=graph-rest-beta)|

>[!NOTE]
> The requests shown in this article uses sample values. You will need update these. The response objects shown may also be shortened for readability. All the properties will be returned from an actual call.

## Step 1: Create an application

### Sign in to Microsoft Graph Explorer (recommended), Postman, or any other API client you use

1. Start [Microsoft Graph Explorer](https://developer.microsoft.com/graph/graph-explorer).
2. Select **Sign-In with Microsoft** and sign in using an Azure AD global administrator or App Admin credentials.
3. Upon successful sign-in, you'll see the user account details in the left pane.

### Create an application

> To configure Application Proxy for an app using the API, you must first create an application, add a service principal to the app, then update the application's **onPremisesPublishing** property to configure the App Proxy settings. When creating the application set the application's **signInAudience** to "AzureADMyOrg".

Reviewer: ```suggestion
To configure Application Proxy for an app, you must create an application, add a service principal to the application, and then update the value of the application's **onPremisesPublishing** property to configure the App Proxy settings. When creating the application set the value of its **signInAudience** property to `AzureADMyOrg`.
```

#### Request

> # [HTTP](#tab/http)

Reviewer: I think that because this tutorial is meant to showcase the REST (HTTP) calls, the other SDK tabs should not be included in the examples. Please remove lines 43 and 56.

  "id": "bf21f7e9-9d25-4da2-82ab-7fdd85049f83",
  "deletedDateTime": null,
  "addIns": [],
  "appId": "d7fbfe28-c60e-46d2-8335-841923950d3b",
  "applicationTemplateId": null,
  "identifierUris": [],
  "createdDateTime": "2020-08-11T21:07:47.5919755Z",
  "description": null,
  "displayName": "Contoso IWA App",
  "isAuthorizationServiceEnabled": false,
  "isDeviceOnlyAuthSupported": null,
  "isFallbackPublicClient": null,
  "groupMembershipClaims": null,
  "notes": null,
  "optionalClaims": null,
  "orgRestrictions": [],
  "publisherDomain": "f128.info",
  "signInAudience": "AzureADandPersonalMicrosoftAccount",
  "tags": [],
  "tokenEncryptionKeyId": null,
  "uniqueName": null,
  "verifiedPublisher": {
      "displayName": null,
      "verifiedPublisherId": null,
      "addedDateTime": null
  },
}
```

### Retrieve the application object ID and appId
Use the response from the previous call to retrieve and save the application object ID and appId.
```
"application": {
  "id": "bf21f7e9-9d25-4da2-82ab-7fdd85049f83",
  "appId": "d7fbfe28-c60e-46d2-8335-841923950d3b"
}
```
### Create a servicePrincipal for the application and add required tags

> Use the **appId** to create a servicePrincipal for the application. Then add the tags required for configuring Application Proxy for an app.

Reviewer: ```suggestion
Use the **appId** to create a servicePrincipal object for the application. After you create the service principal, add the tags required for configuring Application Proxy for an app.
```

> #### Request

Reviewer: The content between lines 129 and 160 seem like a copy and paste error. I think what you want here is just the HTTP code:

POST https://graph.microsoft.com/beta/serviceprincipals
Content-type: appplication/json
{
  "appId":"d7fbfe28-c60e-46d2-8335-841923950d3b",
  "tags": [
    "WindowsAzureActiveDirectoryIntegratedApp",
    "WindowsAzureActiveDirectoryOnPremApp"
  ]
}

Also, see the previous comment about only including the HTTP code.

```http
HTTP/1.1 204 No content
```

> ### Set additional required properties to complete configuring the application

Reviewer: I would try to keep these shorter and more scannable, such as:

### Complete the configuration of the application

>  Update the application's **redirectUri**, **identifierUri**, and **homepageUrl** properties to the external UR configured in onPremisesPublishing property. Then update **[implicitGrantSettings](https://docs.microsoft.com/graph/api/resources/implicitgrantsettings?view=graph-rest-1.0)** to true for enabledTokenIssuance and false for enabledAccessTokenIssuance.

Reviewer: ```suggestion
 Update the value of application's **redirectUri**, **identifierUri**, and **homepageUrl** properties to the external URL that is defined as the value of the **onPremisesPublishing** property. Update the **[implicitGrantSettings](https://docs.microsoft.com/graph/api/resources/implicitgrantsettings?view=graph-rest-1.0)** resource to `true` for the value of the **enabledTokenIssuance** property and to `false` for the value of the **enabledAccessTokenIssuance** property.
```

#### Request

> # [HTTP](#tab/http)

Reviewer: Please remove.

PATCH https://graph.microsoft.com/beta/applications/bf21f7e9-9d25-4da2-82ab-7fdd85049f83
Content-type: appplication/json

{
  "identifierUris": ["https://contosoiwaapp-contoso.msappproxy.net"],
  "web": {
    "redirectUris": ["https://contosoiwaapp-contoso.msappproxy.net"],
    "homePageUrl": "https://contosoiwaapp-contoso.msappproxy.net",
    "implicitGrantSettings": {
      "enableIdTokenIssuance": true,
      "enableAccessTokenIssuance": false
    }
  }
}
```

> # [C#](#tab/csharp)

Reviewer: Please remove these lines.

#### Request

> # [HTTP](#tab/http)

Reviewer: Please remove.

   "name": "IWA Demo Connector Group"
}
```

> # [C#](#tab/csharp)

Reviewer: Please remove these lines.

#### Request

> # [HTTP](#tab/http)

Reviewer: Please remove.

  "@odata.id":"https://graph.microsoft.com/beta/onPremisesPublishingProfiles/applicationProxy/connectorGroups/3e6f4c35-a04b-4d03-b98a-66fff89b72e6"
}
```

> # [C#](#tab/csharp)

Reviewer: Please remove these lines.

#### Request

> # [HTTP](#tab/http)

Reviewer: Please remove.

   }
}
```

> # [C#](#tab/csharp)

Reviewer: Please remove these lines.

  "name": "servicePrincipals"
}-->
```msgraph-interactive
GET https://graph.microsoft.com/beta/servicePrincipals/a8cac399-cde5-4516-a674-819503c61313/appRoles
```

> # [C#](#tab/csharp)

Reviewer: Please remove.

| principalId | User ID of the user that will be assigned to the app | 2fe96d23-5dc6-4f35-8222-0426a8c115c8 |
| principalType | Type of user | User |
| appRoleId |  The App role ID of the default app role of the app | 18d14569-c3bd-439b-9a66-3a2aee01d14f |
| resourceId | The servicePrincipal ID of the app | a8cac399-cde5-4516-a674-819503c61313 |

#### Request

> # [HTTP](#tab/http)

Reviewer: Please remove.

  "principalId": "2fe96d23-5dc6-4f35-8222-0426a8c115c8",
  "principalType": "User",
  "appRoleId":"18d14569-c3bd-439b-9a66-3a2aee01d14f",
  "resourceId":"a8cac399-cde5-4516-a674-819503c61313"
}

> ```

Reviewer: Please remove these lines.
